### PR TITLE
WIP: Dockerify everything

### PR DESCRIPTION
**Pull request by msmith-techempower**
The toolset is invoked in a container, wrk is run from a container, the framework implementations are run from within a container, and the databases are run in a container.

How to run locally:
1) Docker must be installed on host machine installed
2) `./tfb.sh --mode verify --test gemini`
2a) On Windows hosts, you will need to have ubuntu subsystem installed (or Cygwin, etc) and a variable to run: `TFB_SOURCE_DIR=c:/Path/to/FrameworkBenchmarks ./tfb.sh --mode verify --test gemini`

### CODE REVIEW

File: toolset/setup/docker/wrk.dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+##############################
+RUN curl -sL -o wrk-4.0.1.tar.gz https://github.com/wg/wrk/archive/4.0.1.tar.gz
+RUN tar xzf wrk-4.0.1.tar.gz
+RUN cd wrk-4.0.1 && make
+RUN cp wrk-4.0.1/wrk /usr/local/bin
+
+#############################
+# pipeline.lua
+#############################
+RUN echo "init = function(args)" >> pipeline.lua; \
```
[the remark this answers is not in the document]
Comment (msmith-techempower) [reply toolset/setup/docker/wrk.dockerfile]: Because I'm dumb, and it should definitely be done that way.

File: tfb.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+docker run --network=host -v //var/run/docker.sock:/var/run/docker.sock --mount type=bind,source=c:/Development/FrameworkBenchmarks,target=/FrameworkBenchmarks tfb "$@"
```
Comment (RX14) [inline tfb.sh]: Is `--net=host` really required here? Also there's two `/` in `//var`, plus the whole `--mount` option can be shortened to `` -v `pwd`:/FrameworkBenchmarks``

File: toolset/setup/docker/wrk/concurrency.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+echo ""
+wrk "-H 'Host: $server_host' -H 'Accept: $accept' -H 'Connection: keep-alive'" --latency -d $duration -c $max_concurrency --timeout 8 -t $max_threads $url
+sleep 5
+
+echo ""
+echo "---------------------------------------------------------"
+echo " Synchronizing time"
+echo "---------------------------------------------------------"
+echo ""
+ntpdate -s pool.ntp.org
```
Comment (RX14) [inline toolset/setup/docker/wrk/concurrency.sh]: This shouldn't work in docker. Neither should it be neccesary.

File: toolset/setup/docker/wrk/concurrency.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+echo ""
+echo "---------------------------------------------------------"
+echo " Concurrency: $c for $name"
+echo " wrk \"-H 'Host: $server_host' -H 'Accept: $accept' -H 'Connection: keep-alive'\" --latency -d %duration -c $c --timeout 8 -t $(($c>$max_threads?$max_threads:$c)) \"$url\""
+echo "---------------------------------------------------------"
+echo ""
+STARTTIME=$(date +"%s")
+wrk "-H 'Host: $server_host' -H 'Accept: $accept' -H 'Connection: keep-alive'" --latency -d $duration -c $c --timeout 8 -t "$(($c>$max_threads?$max_threads:$c))" $url
+echo "STARTTIME $STARTTIME"
+echo "ENDTIME $(date +"%s")"
```
Comment (RX14) [inline toolset/setup/docker/wrk/concurrency.sh]: I'd trust `wrk`'s accounting of time more than doing this.

**Comment by nbrady-techempower** [comment]
Sadly any docker commands in the build script route to a `sudo: required` travis container

image

**Pull request merged**